sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp

```diff
@@ -67,10 +67,27 @@ struct joint_matrix;
 
 } // namespace matrix
 } // namespace experimental
+
+namespace detail {
+// Differentiating between the "element type" and the "storage element type"
+template <typename T> struct helper_traits {
```

Comment:
Since this is in a very general namespace (`sycl::ext::oneapi::detail`) it would be better to have the name be a little more specific. What is it a helper-trait for?

Reply:
We can call it something like jm_type_interpretation_helper_trait.

Reply:
Sounds good to me! Definitely seems to remove most ambiguity and decreases the chance of name conflicts.